src/yaml-support/yaml-schema.ts

```diff
@@ -0,0 +1,165 @@
+        if (node[KUBERNETES_GROUP_VERSION_KIND]) {
+            let groupKindNode = node[KUBERNETES_GROUP_VERSION_KIND];
+            delete node[KUBERNETES_GROUP_VERSION_KIND];
+            if (groupKindNode.length) {
+                groupKindNode.forEach((groupKindNodeItem) => {
+                    const apiVersion = ((groupKindNodeItem.Group ? groupKindNodeItem.Group + '/' : '') + groupKindNodeItem.Version);
+                    const kind = groupKindNodeItem.Kind;
+                    _definitions[apiVersion + GROUP_VERSION_KIND_SEPARATOR + kind] = _definitions[name.toLowerCase()] = {
+                        apiVersion,
+                        name,
+                        kind,
+                        ...node
+                    };
+
+
+                });
+            }
+        } else {
+            _definitions[name.toLowerCase()] = {
+                name,
+                ...node
+            };
+        }
+
+        toFixed.push(node['properties']);
+    }
+
+    for (let fix of toFixed ) {
+        patchOnRef(fix);
+    }
+}
+
+export async function registerYamlSchemaSupport() {
+    loadSchema(KUBERNETES_SCHEMA);
+    let yamlPlugin;
+    const installedExtensions: any[] = vscode.extensions.all;
+    for (let ext of installedExtensions) {
+        if (ext.id === VSCODE_YAML) {
+            try {
+                yamlPlugin = await Promise.resolve(ext.activate());
+                break;
+            } catch (error) {
+                console.log('Failed to activate the Yaml Extension: ' + error);
+            }
+            break;
+        }
+    }
+    if (!yamlPlugin) {
+        console.log('vscode-yaml plugin doesn\'t expose the entry to register custom schema provider.');
+        return;
+    }
+    // register for kubernetes schema provider
+    yamlPlugin.registerContributor('kubernetes', (resource) => {
+        const textEditor = vscode.window.visibleTextEditors.find((editor) => editor.document.uri.toString() === resource);
+        if (textEditor) {
+            const yamlDocs = yamlLocator.getYamlDocuments(textEditor.document);
+            let choices: string[] = [];
+            yamlDocs.forEach((doc)=> {
+                // if the yaml document contains apiVersion and kind node, it will report it is a kubernetes yaml
+                // file
+                const rootMapping = doc.nodes.find((node) => node.kind === 'MAPPING');
+                // TODO, unwrap quotes
+                const apiVersion = rootMapping.mappings.find((map) => map.key && map.key.raw === 'apiVersion');
+                const kind = rootMapping.mappings.find((map) => map.key && map.key.raw === 'kind');
+                if (apiVersion && kind) {
+                    choices.push(apiVersion.value.raw + GROUP_VERSION_KIND_SEPARATOR + kind.value.raw);
+                }
+            });
+            choices = _.uniq(choices);
+            if (choices.length > 1) {
+                return 'kubernetes://combine/' + _.uniq(choices).join('+');
```

Comment:
I don't see this URL prefix referenced elsewhere.  Is it understood by the YAML extension by virtue of the `kubernetes` contributor being registered, and `combine` being a well-known keyword?  If so, you need to make it clearer that the `kubernetes` registration key and the URL scheme are related.

Reply:
it is useless for the phase combine, just a placeholder for uri host, we want the schema name as the `path` part rather than the host part.